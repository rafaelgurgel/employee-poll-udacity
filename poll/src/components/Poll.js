import React from 'react'
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'

function getAvatar(avatarURL) {
  return avatarURL ? avatarURL : 'https://via.placeholder.com/50'
}

export default function Poll({ id }) {
  const { question, author } = useSelector((state) => {
    const question = state.questions[id];
    const author = question ? state.users[question.author] : null;
    return { question, author };
  });

  if (!question) {
    return <p>This question doesn't exist</p>;
  }

  if (!author) {
    return <p>Author not found</p>;
  }

  return (
    <div className="poll" style={{ border: '1px solid #ccc', margin: '1rem 0', padding: '1rem' }}>
      <div style={{ display: 'flex', alignItems: 'center', gap: '1rem' }}>
        <img
          src={getAvatar(author.avatarURL)}
          alt={`Avatar of ${author.name}`}
          width="40"
          height="40"
          style={{ borderRadius: '50%' }}
        />
        <h4>{author.name} asks:</h4>
      </div>
      <p style={{ marginTop: '1rem' }}>
        Would you rather <strong>{question.optionOne.text}</strong> or ...
      </p>
      <Link to={`/questions/${id}`}>View Poll</Link>
    </div>
  );
}
