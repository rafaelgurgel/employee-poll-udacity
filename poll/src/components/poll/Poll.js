// src/components/Poll.js
import React from 'react'
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import getAvatar from '../../utils/getAvatar'

import './poll.css'

export default function Poll({ id }) {
  const { question, author } = useSelector((state) => {
    const question = state.questions[id]
    const author = question ? state.users[question.author] : null
    return { question, author }
  })

  if (!question) {
    return <p>This question doesn't exist</p>
  }

  if (!author) {
    return <p>Author not found</p>
  }

  return (
    <div className="poll-container">
      <div className="poll-header">
        <img
          src={getAvatar(author.avatarURL)}
          alt={`Avatar of ${author.name}`}
          className="poll-avatar"
        />
        <h4>{author.name} asks:</h4>
      </div>
      <p className="poll-content">
        Would you rather <strong>{question.optionOne.text}</strong> or ...
      </p>
      <Link to={`/questions/${id}`} className="poll-link">
        View Poll
      </Link>
    </div>
  )
}
