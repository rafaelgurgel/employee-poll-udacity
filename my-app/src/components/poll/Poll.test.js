import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Poll from './Poll'
import '@testing-library/jest-dom'

const mockReducer = (state = {
  questions: {
    q1: {
      id: 'q1',
      author: 'sarahedo',
      optionOne: { votes: [], text: 'OptionOne' },
      optionTwo: { votes: [], text: 'OptionTwo' }
    },
  },
  users: {
    sarahedo: { id: 'sarahedo', name: 'Sarah Edo', avatarURL: null }
  }
}) => state

const mockStore = createStore(mockReducer)

describe('Poll Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <Poll id="q1" />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
