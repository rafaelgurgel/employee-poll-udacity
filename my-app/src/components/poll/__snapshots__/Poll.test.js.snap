// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="poll-container"
  >
    <div
      class="poll-header"
    >
      <img
        alt="Avatar of Sarah Edo"
        class="poll-avatar"
        src="defaultAvatar.svg"
      />
      <h4>
        Sarah Edo asks:
      </h4>
    </div>
    <p
      class="poll-content"
    >
      Would you rather 
      <strong>
        OptionOne
      </strong>
       or ...
    </p>
    <a
      class="poll-link"
      href="/questions/q1"
    >
      View Poll
    </a>
  </div>
</DocumentFragment>
`;
